### PR TITLE
Content excpetion handling in GenerateXmlElements function

### PR DESCRIPTION
**Pull request by adnanzameer**
Handling of failed generating XML sitemap

"Geta.SEO.Sitemaps.XML.SitemapXmlGenerator : Error on generating xml sitemap
EPiServer.Core.PageNotFoundException: Content with id XXXXX was not found"

As IContentRepository TryGet actually throws an exception when there is something wrong.  GenerateXmlElements function's for-each loop breaks and failed the sitemap generation job

### CODE REVIEW

File: src/Geta.SEO.Sitemaps/XML/SitemapXmlGenerator.cs [hunk 1]
```diff
@@ -564,5 +562,39 @@ protected bool IsAbsoluteUrl(string url, out Uri absoluteUri)
...
+            }
+            catch (Exception e)
+            {
+                Log.Error("Error on contentReference " + contentLink.ID + Environment.NewLine + e);
+            }
+
+            return false;
+        }
+
+        protected bool TryGet<T>(ContentReference contentLink, LoaderOptions settings, out T content) where T : IContentData
```
Comment (patkleef) [inline src/Geta.SEO.Sitemaps/XML/SitemapXmlGenerator.cs:583]: Could you merge these two methods? And make LoaderOptions an optional parameter?

**Review by patkleef — CHANGES_REQUESTED** [review]
Thank you for the contribution, added a comment in your PR.

Also could you change the target branch to develop? So we test out the change before we merge it to master.

File: src/Geta.SEO.Sitemaps/XML/SitemapXmlGenerator.cs (hunk 1 above)
Comment (adnanzameer) [reply src/Geta.SEO.Sitemaps/XML/SitemapXmlGenerator.cs:571]: I think I manged to reduce the code

**Comment by adnanzameer** [comment]
> Thank you for the contribution, added a comment in your PR.
> 
> Also could you change the target branch to develop? So we test out the change before we merge it to master.

@patkleef all done btw

**Review by patkleef — APPROVED** [review]

**Pull request merged**